pkgs/by-name/sd/SDL2_sound/package.nix

```diff
@@ -2,53 +2,52 @@
   stdenv,
   lib,
   fetchFromGitHub,
-  fetchpatch,
   cmake,
   SDL2,
-  flac,
-  libmikmod,
-  libvorbis,
-  timidity,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "SDL2_sound";
-  version = "2.0.1";
+  version = "2.0.4";
 
   src = fetchFromGitHub {
     owner = "icculus";
     repo = "SDL_sound";
-    rev = "v${version}";
-    hash = "sha256-N2znqy58tMHgYa07vEsSedWLRhoJzDoINcsUu0UYLnA=";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-5t2ELm8d8IX+cIJqGl/8sffwXGj5Cm0kZI6+bmjvvPg=";
   };
 
-  patches = [
-    (fetchpatch {
-      # https://github.com/icculus/SDL_sound/pull/32 - fix build on darwin
-      # can be dropped on the next update
-      url = "https://github.com/icculus/SDL_sound/commit/c15d75b7720113b28639baad284f45f943846294.patch";
-      hash = "sha256-4GL8unsZ7eNkzjLXq9QdaxFQMzX2tdP0cBR1jTaRLc0=";
-    })
+  outputs = [
+    "out"
+    "lib"
+    "dev"
   ];
 
   nativeBuildInputs = [ cmake ];
 
-  cmakeFlags = [ "-DSDLSOUND_DECODER_MIDI=1" ];
-
-  buildInputs = [
-    SDL2
-    flac
-    libmikmod
-    libvorbis
-    timidity
+  cmakeFlags = [
+    (lib.cmakeBool "SDLSOUND_DECODER_MIDI" true)
+    (lib.cmakeBool "SDLSOUND_BUILD_SHARED" (!stdenv.hostPlatform.isStatic))
+    (lib.cmakeBool "SDLSOUND_BUILD_STATIC" stdenv.hostPlatform.isStatic)
   ];
 
-  meta = with lib; {
+  buildInputs = [ SDL2 ];
+
+  meta = {
     description = "SDL2 sound library";
     mainProgram = "playsound";
-    platforms = platforms.unix;
-    license = licenses.zlib;
+    platforms = lib.platforms.all;
+    license = with lib.licenses; [
+      zlib
+
+      # various vendored decoders
+      publicDomain
+
+      # timidity
+      artistic1
+      lgpl21Only
```

Comment:
How viable would it be to unvendor timidity and link ours into SDL2_sound? Typically unvendoring should be preferred if possible.